src/widgets/ActivityGroups/model/hooks/useTakeNowRedirect.ts

```diff
@@ -0,0 +1,87 @@
+import { useMemo } from 'react';
+
+import { openStoreLink } from '~/abstract/lib';
+import { isSupportedActivity } from '~/entities/activity';
+import { appletModel } from '~/entities/applet';
+import { useStartSurvey } from '~/features/PassSurvey';
+import { AppletBaseDTO, AppletEventsResponse, HydratedAssignmentDTO } from '~/shared/api';
+
+export function useTakeNowRedirect({
+  activityOrFlowId,
+  applet,
+  assignments = [],
+  events = [],
+  isPublic = false,
+  publicAppletKey = null,
+}: {
+  activityOrFlowId?: string | null;
+  applet?: AppletBaseDTO;
+  assignments?: HydratedAssignmentDTO[] | null;
+  events?: AppletEventsResponse['events'];
+  isPublic?: boolean;
+  publicAppletKey?: string | null;
+}) {
+  const { startSurvey } = useStartSurvey({ applet, isPublic, publicAppletKey });
+  const { getMultiInformantState } = appletModel.hooks.useMultiInformantState();
+
+  const handleTakeNowRedirect = useMemo(
+    () =>
+      ({
+        activityId,
+        eventId,
+        flowId = null,
+        isSupported = true,
+        targetSubjectId = null,
+      }: {
+        activityId: string;
+        eventId: string;
+        flowId?: string | null;
+        isSupported?: boolean;
+        targetSubjectId?: string | null;
+      }) => {
+        if (!isSupported) {
+          openStoreLink();
+          return;
+        }
+
+        startSurvey({ activityId, eventId, flowId, shouldRestart: true, targetSubjectId });
```

Comment:
This is the culprit causing the "About …" labels to be shown in the assessment. This parameter was in fact added strictly for MI Assignment, not for Take Now, while Take Now uses a different mechanism for tracking target subject (and source subject) IDs.

The parameter name below could have been named more clearly (like `assignmentTargetSubjectId`), but alas!

So I would remove all references to `targetSubjectId` within this hook, and pass null here, and you should be good to go!

```suggestion
        startSurvey({ activityId, eventId, flowId, shouldRestart: true, targetSubjectId: null });
```

Reply:
Updated in [this commit](https://github.com/ChildMindInstitute/mindlogger-web-refactor/pull/536/commits/3941a9fec1cd48ea74b54625377170cacc364ab6)!

Reply:
Tested and looks like it's working!